aws/resource_aws_s3_bucket_analytics_configuration_test.go

```diff
@@ -0,0 +1,1225 @@
+  bucket = "%s"
+}
+`, name, prefix, bucket)
+}
+
+func testAccAWSS3BucketAnalyticsConfigurationWithFilterSingleTag(name, bucket, tag string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket_analytics_configuration" "test" {
+  bucket = aws_s3_bucket.test.bucket
+  name   = "%s"
+
+  filter {
+    tags = {
+      "tag1" = "%s"
+    }
+  }
+}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+}
+`, name, tag, bucket)
+}
+
+func testAccAWSS3BucketAnalyticsConfigurationWithFilterMultipleTags(name, bucket, tag1, tag2 string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket_analytics_configuration" "test" {
+  bucket = aws_s3_bucket.test.bucket
+  name   = "%s"
+
+  filter {
+    tags = {
+      "tag1" = "%s"
+      "tag2" = "%s"
+    }
+  }
+}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+}
+`, name, tag1, tag2, bucket)
+}
+
+func testAccAWSS3BucketAnalyticsConfigurationWithFilterPrefixAndTags(name, bucket, prefix, tag1, tag2 string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket_analytics_configuration" "test" {
+  bucket = aws_s3_bucket.test.bucket
+  name   = "%s"
+
+  filter {
+    prefix = "%s"
+
+    tags = {
+      "tag1" = "%s"
+      "tag2" = "%s"
+    }
+  }
+}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+}
+`, name, prefix, tag1, tag2, bucket)
+}
+
+func testAccAWSS3BucketAnalyticsConfigurationWithEmptyStorageClassAnalysis(name, bucket string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket_analytics_configuration" "test" {
+  bucket = aws_s3_bucket.test.bucket
+  name   = "%s"
+
+  storage_class_analysis {
+  }
+}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+}
+`, name, bucket)
+}
+
+func testAccAWSS3BucketAnalyticsConfigurationWithDefaultStorageClassAnalysis(name, bucket string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket_analytics_configuration" "test" {
+  bucket = aws_s3_bucket.test.bucket
+  name   = "%s"
+
+  storage_class_analysis {
+    data_export {
+      destination {
+        s3_bucket_destination {
+          bucket_arn = aws_s3_bucket.destination.arn
+        }
+      }
+    }
+  }
+}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+}
+
+resource "aws_s3_bucket" "destination" {
+  bucket = "%s-destination"
+}
+`, name, bucket, bucket)
+}
+
+func testAccAWSS3BucketAnalyticsConfigurationWithFullStorageClassAnalysis(name, bucket, prefix string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket_analytics_configuration" "test" {
+  bucket = aws_s3_bucket.test.bucket
+  name   = "%s"
+
+  storage_class_analysis {
+    data_export {
+      output_schema_version = "V_1"
+      destination {
+        s3_bucket_destination {
+          format     = "CSV"
+          bucket_arn = aws_s3_bucket.destination.arn
+          prefix     = "%s"
+        }
+      }
+    }
+  }
+}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+}
+
+resource "aws_s3_bucket" "destination" {
+  bucket = "%s-destination"
+}
+`, name, prefix, bucket, bucket)
```

Comment:
Nit: To save repeating fmt arguments, you can use indexed notation in the template, e.g. `%[3]s`